### PR TITLE
createBatch command for signing entire folder

### PR DESCRIPTION
**Pull request by metsma**
Signed-off-by: Raul Metsma <raul@metsma.ee>

### CODE REVIEW

File: src/crypto/PKCS11Signer.cpp [hunk 1]
```diff
@@ -327,7 +327,8 @@ vector<unsigned char> PKCS11Signer::sign(const string &method, const vector<unsi
         }
         switch(rv)
         {
-        case CKR_OK: break;
```
Comment (martinpaljak) [inline src/crypto/PKCS11Signer.cpp]: This leads to bad readability and is error-prone. The fallthrough can stay, but should be made more explicit.

File: src/digidoc-tool.cpp [hunk 2]
```diff
@@ -830,6 +836,73 @@ static int create(int argc, char* argv[])
...
 
 /**
+ * Create new container.
+ *
+ * @param argc number of command line arguments.
+ * @param argv command line arguments.
+ * @return EXIT_FAILURE (1) - failure, EXIT_SUCCESS (0) - success
+ */
+static int createPatch(int argc, char* argv[])
+{
```
Comment (Telgat) [inline src/digidoc-tool.cpp]: This method does a lot of the same things as signContainer(), as in everything except the for loop, the .asice check and the dontSign flag check, that means a whole lot of code duplication here :(
Could just use the call to signContainer() inside the loop or extract the common parts to separate methods

File: src/digidoc-tool.cpp [hunk 3]
```diff
@@ -316,6 +316,11 @@ static void printUsage(const char *executable)
     << "      --mime=        - can be after --file parameter. Default value is application/octet-stream" << endl
     << "      --dontsign     - Don't sign the newly created container." << endl
     << "      for additional options look sign command" << endl << endl
+    << "  Command createPatch:" << endl
```
Comment (Telgat) [inline src/digidoc-tool.cpp]: Could perhaps use a clearer name, it's probably only programmers who associate patch with actually patching something (no idea what normal people think though), but I would probably prefer something like "signAllFilesInFolder" but then again that could mislead to believe that all the files end up inside the same container...

**Comment by metsma** [comment]
Update commit

File: src/digidoc-tool.cpp [hunk 4]
```diff
@@ -830,6 +844,51 @@ static int create(int argc, char* argv[])
...
+    int returnCode = EXIT_SUCCESS;
+    for(const string &file: File::listFiles(p.path))
+    {
+        if(file.compare(file.size() - 6, 6, ".asice") == 0)
+            continue;
+        std::cout << "Signing file: " << file << endl;
+        try {
+            unique_ptr<Container> doc(Container::create(file + ".asice"));
+            doc->addDataFile(file, "application/octet-stream");
+            if(signContainer(doc.get(), signer.get(), p.dontValidate) == EXIT_FAILURE)
```
Comment (metsma) [inline src/digidoc-tool.cpp]: Default is int returnCode = EXIT_SUCCESS;
and turn total status failure only then when one if(signContainer == EXIT_FAILURE)" fails.
It does not terminate process when some signatures are invalid

**Pull request merged**